clock_exercise/clockmanager.cpp

```diff
@@ -0,0 +1,47 @@
+#include <QTime>
+#include <memory>
+#include<QDebug>
+#include "clockmanager.h"
+
+
+
+ClockManager::ClockManager(QObject *parent) : QObject(parent), mTimerForRefresh(new QTimer())
+{
+    connect(mTimerForRefresh.get(), &QTimer::timeout, this, &ClockManager::refresh);
+    mTimerForRefresh->start(1000);
+
+    refresh();
+}
+
+void ClockManager::refreshDateText() {
+    QString previousDateText = mDateText;
```

Comment:
Secondo me qui potresti:
* assegnare a `QString previousDateText` direttamente la nuova data
* se è uguale alla vecchia chiamare `return`
* fare successivamente l'assegnamento e l'`emit`

Così ti eviti una copia, e in più logicamente mi tornerebbe vedere l'assegnamento accorpato all'`emit`